model/task/task.go

```diff
@@ -3709,6 +3709,67 @@ func GetGroupedTaskStatsByVersion(versionID string, opts GetTasksByVersionOption
 
 }
 
+// GetMatchingBaseTaskStatuses returns the corresponding base task statuses for tasks on a version.
+// If a task ran on the base version but was not scheduled on the version, then its status won't be returned.
+func GetMatchingBaseTaskStatuses(versionID string, baseVersionID string) ([]string, error) {
+	pipeline := []bson.M{}
+	taskField := "tasks"
+
+	// Fetch all ACTIVATED tasks from version, and all tasks from base version.
+	pipeline = append(pipeline, bson.M{
+		"$match": bson.M{
+			"$or": []bson.M{
+				{VersionKey: baseVersionID},
+				{VersionKey: versionID, ActivatedTimeKey: bson.M{"$ne": utility.ZeroTime}},
+			},
+		}})
+	// Add display status, then group by display name and build variant
+	pipeline = append(pipeline, addDisplayStatus)
```

Comment:
fyi known issues won't be surfaced here because the known-issue status is never added before this step.

Reply:
Oh yeah, I see what you mean, because it doesn't run the annotation facet 😓

But I think it might actually be impossible for any base status in the Tasks table to show as `Known Issue`, since we never run the [annotation facet](https://github.com/evergreen-ci/evergreen/blob/main/model/task/task.go#L3837) directly on the base tasks. I tried it out in Compass to see if the facet + setting ROOT affects the Collection in `$lookup` but it seems like it doesn't.

Since it seems like there'd be no real effect of adding the annotations, I'll merge this as is.